engine/src/main/java/org/terasology/telemetry/TelemetryComponentSystem.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.telemetry;
+
+import com.snowplowanalytics.snowplow.tracker.emitter.Emitter;
+import com.snowplowanalytics.snowplow.tracker.events.Unstructured;
+import org.terasology.config.Config;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.registry.In;
+import org.terasology.telemetry.metrics.ModulesMetric;
+
+/**
+ * This component system is used to track metrics in game.
```

Comment:
Is this meant to be a new template for systems to extend? Or the single system that tracks all metrics?

Maybe this is just personal interpretation, but I thought of "BaseComponentSystem" as sort of a generic type name. Like "This includes some basics a general component system might like". Seeing this get named "TelemetryComponentSystem" makes me wonder if it is defining a new template for Systems, rather than a single specific thing.

If this system is purely tracking metrics and no other systems like it are expected I'd suggest simply naming it "TelemetrySystem" or something like that :-)